packages/mdx/mdx-hast-to-jsx.js

```diff
@@ -47,7 +52,13 @@ function toJSX(node, parentNode = {}, options = {}) {
 
   // recursively walk through children
   if (node.children) {
-    children = node.children.map(childNode => toJSX(childNode, node)).join('')
+    children = node.children.map(childNode => {
+      const childOptions = {
+        ...options,
+        preserveNewlines: preserveNewlines || node.tagName === 'pre',
```

Comment:
👍 I think this makes sense, as long as there's information about this default behavior somewhere.